source/draw/gpu/opengl/OpenGLContext.ooc

```diff
@@ -105,6 +105,9 @@ OpenGLContext: class extends GpuContext {
 	_searchImageBin: func (type: Class, size: IntVector2D) -> GpuImage {
 		this _recycleBin search(|image| image instanceOf(type) && image size == size)
 	}
+	createFence: override func -> Promise {
+		OpenGLPromise new(this)
```

Comment:
`OpenGLPromise` is a promise that can synchronize renderings on the GPU, but the idea is that we don't need to know what kind of Promise it is when we wait for it.

```ooc
createFence: override func -> Promise {
    result := OpenGLPromise new()
    result sync()
    result
}

//Using it:
promise := context createFence()
promise wait() . free()